pkg/capabilities/consensus/ocr3/reporting_plugin.go

```diff
@@ -200,31 +198,39 @@ func (r *reportingPlugin) Reports(seqNr uint64, outcome ocr3types.Outcome) ([]oc
 
 	reports := []ocr3types.ReportWithInfo[[]byte]{}
 
-	// This doesn't handle a query which contains the same workflowId multiple times.
-	for _, report := range o.ReportsToGenerate {
+	for _, report := range o.CurrentReports {
 		outcome, id := report.Outcome, report.Id
-		outcome, err := pbtypes.AppendWorkflowIDs(outcome, id.WorkflowId, id.WorkflowExecutionId)
-		if err != nil {
-			r.lggr.Errorw("could not append IDs")
-			continue
-		}
 
-		enc, err := r.r.getEncoder(id.WorkflowId)
-		if err != nil {
-			r.lggr.Errorw("could not retrieve encoder for workflow", "error", err, "workflowID", id.WorkflowId)
-			continue
+		info := &pbtypes.ReportInfo{
+			Id:           id,
+			ShouldReport: outcome.ShouldReport,
 		}
 
-		mv := values.FromMapValueProto(outcome.EncodableOutcome)
-		report, err := enc.Encode(context.Background(), *mv)
-		if err != nil {
-			r.lggr.Errorw("could not encode report for workflow", "error", err, "workflowID", id.WorkflowId)
-			continue
+		var report []byte
+		if info.ShouldReport {
+			newOutcome, err := pbtypes.AppendWorkflowIDs(outcome, id.WorkflowId, id.WorkflowExecutionId)
+			if err != nil {
+				r.lggr.Errorw("could not append IDs")
+				continue
+			}
+
+			enc, err := r.r.getEncoder(id.WorkflowId)
+			if err != nil {
+				r.lggr.Errorw("could not retrieve encoder for workflow", "error", err, "workflowID", id.WorkflowId)
+				continue
+			}
+
+			mv := values.FromMapValueProto(newOutcome.EncodableOutcome)
+			report, err = enc.Encode(context.Background(), *mv)
```

Comment:
This feels a little risky. We require every aggregator to always produce an encodable report even if should_report was set to true. It will work with DataFeeds (empty array) but is it generally safe?
It's especially weird given that ShouldAccept later ignores the encoded report altogether :) 

I'm not sure what an alternative could be, though... a signed empty byte array (with attached IDs)...?

Reply:
ugh, of course, sorry for a silly comment